tests/smartcam/modules/test_pantilt.py

```diff
@@ -0,0 +1,141 @@
+"""Tests for PanTilt module."""
+
+from __future__ import annotations
+
+import pytest
+from pytest_mock import MockerFixture
+
+from kasa import Device
+from kasa.smartcam.modules.pantilt import PanTilt
+
+from ...conftest import device_smartcam
+
+
+@device_smartcam
+async def test_pantilt(dev: Device, mocker: MockerFixture):
+    """Test PanTilt module."""
+    pantilt = dev.modules.get("PanTilt")
+    if pantilt is None:
+        pytest.skip("Device does not have PanTilt module")
+
+    assert isinstance(pantilt, PanTilt)
+
+    # Check if device has presets from fixture
+    if pantilt._presets:
+        # Preset feature should be available
+        assert "preset" in dev.features
+
+        preset_feature = dev.features["preset"]
+        assert preset_feature is not None
+
+        # Get first preset name
+        first_preset_name = next(iter(pantilt._presets.keys()))
+        assert preset_feature.value == first_preset_name
+
+        # Mock the protocol query for testing set_value
+        # This allows set_preset function body to be executed (lines 110-112)
+        mock_protocol_query = mocker.patch.object(dev.protocol, "query")
+        mock_protocol_query.return_value = {}
+
+        # Set to a valid preset - this executes set_preset function (lines 109-112)
+        await preset_feature.set_value(first_preset_name)
+
+        # Verify goto_preset was called with correct preset_id
+        mock_protocol_query.assert_called_once()
+        call_args = mock_protocol_query.call_args
+        assert "motor" in str(call_args) or "preset" in str(call_args).lower()
+
+        # Reset mock
+        mock_protocol_query.reset_mock()
+
+        # Test save_preset
+        await pantilt.save_preset("NewPreset")
+
+        # Note: The Tapo API has a typo in the method name (addMotorPostion instead of addMotorPosition)
+        mock_protocol_query.assert_called_with(
+            request={
+                "addMotorPostion": {
+                    "preset": {"set_preset": {"name": "NewPreset", "save_ptz": "1"}}
+                }
+            }
+        )
+
+
+@device_smartcam
+async def test_pantilt_no_presets(dev: Device, mocker: MockerFixture):
+    """Test PanTilt module behavior when no presets are configured."""
+    pantilt = dev.modules.get("PanTilt")
+    if pantilt is None:
+        pytest.skip("Device does not have PanTilt module")
```

Comment:
```suggestion
```
Instead of doing this check for every separate test, it would be cleaner to define a new parametrization and then use it instead of `@device_smartcam`, for inspiration see:
https://github.com/python-kasa/python-kasa/blob/347bf9a419d20089227cc4b9eaa2af257ef32b82/tests/smartcam/modules/test_childsetup.py#L12-L18